solarforecastarbiter/metrics/deterministic.py

```diff
@@ -19,218 +19,276 @@
 
     """
 
-    return np.mean(y_pred - y_true)
+    return np.mean(fx - obs)
 
 
-def root_mean_square(y_true, y_pred):
+def root_mean_square(obs, fx):
     """Root mean square error (RMSE).
 
+        RMSE = sqrt( 1/n sum_{i=1}^n (fx_i - obs_i)^2 )
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
 
     Returns
     -------
     rmse : float
-        The RMSE between the true and predicted values.
+        The RMSE of the forecast.
 
     """
 
-    return np.sqrt(np.mean((y_true - y_pred) ** 2))
+    return np.sqrt(np.mean((fx - obs) ** 2))
 
 
-def mean_absolute_percentage(y_true, y_pred):
+def mean_absolute_percentage(obs, fx):
     """Mean absolute percentage error (MAPE).
 
+        MAPE = 1/n sum_{i=1}^n |(fx_i - obs_i) / obs_i| * 100%
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
 
     Returns
     -------
     mape : float
-        The MAPE [%] between the true and predicted values.
+        The MAPE [%] of the forecast.
 
     """
 
-    return np.mean(np.abs((y_true - y_pred) / y_true)) * 100.0
+    return np.mean(np.abs((obs - fx) / obs)) * 100.0
 
 
-def normalized_root_mean_square(y_true, y_pred, y_norm):
+def normalized_root_mean_square(obs, fx, norm):
     """Normalized root mean square error (NRMSE).
 
+        NRMSE = RMSE / norm * 100%
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
-    y_norm : float
-        Normalized factor, in the same units as y_true and y_pred.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
+    norm : float
+        Normalized factor, in the same units as obs and fx.
 
     Returns
     -------
     nrmse : float
-        The NRMSE [%] between the true and predicted values.
+        The NRMSE [%] of the forecast.
 
     """
 
-    return root_mean_square(y_true, y_pred) / y_norm * 100.0
+    return root_mean_square(obs, fx) / norm * 100.0
 
 
-def forecast_skill(y_true, y_pred, y_ref):
+def forecast_skill(obs, fx, ref):
     """Forecast skill (s).
 
+        s = 1 - RMSE_fx / RMSE_ref
+
+    where RMSE_fx is the RMSE of the forecast and RMSE_ref is the RMSE of the
+    reference forecast (e.g. Persistence).
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
-    y_ref: array_like
-        Reference forecast values.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
+    ref : (n,) array_like
+        A reference forecast.
```

Comment:
mild preference for the original wording. why the change?

Reply:
To clarify: do you mean you prefer "Reference forecast values." instead of "A reference forecast" or are you referring to the previous "True values.", "Predicted values.", etc. in all the docstrings?

If the latter, my thought was that it's better to use "observation" with `obs` and "forecast" with `fx` to reinforce the variable naming choice. But I'm fine keeping the old wording if that's preferred. (I don't have a strong preference either way.)

Reply:
> do you mean you prefer "Reference forecast values" instead of "A reference forecast"

This but not a big deal. Seems more consistent with the other doc strings

Reply:
Gotcha. I'll revise to `Reference forecast values`. (I agree that sounds better and is more consistent.)